scripts/snapshot.py

```diff
@@ -83,42 +81,41 @@ def validate_options(options):
     options.source_repo = apparent_source_repo_type
   else:
     if options.source_repo != "none" and options.source_repo != apparent_source_repo_type:
-      raise RuntimeError("Specified source repository type of %s conflicts with determined type of %s" % \
-        (options.source_repo, apparent_source_repo_type))
+      raise RuntimeError(f"Specified source repository type of {options.source_repo} conflicts with determined type of {apparent_source_repo_type}")
 
   if options.dest_repo == "":
     #destination repo type is not specified to just using the apparent type.
     options.dest_repo = apparent_dest_repo_type
   else:
     if options.dest_repo != "none" and options.dest_repo != apparent_dest_repo_type:
-      raise RuntimeError("Specified destination repository type of %s conflicts with determined type of %s" % \
-        (options.dest_repo, apparent_dest_repo_type))
+      raise RuntimeError(f"Specified destination repository type of {options.dest_repo} conflicts with determined type of {apparent_dest_repo_type}")
 
   return options
 #end validate_options
 
 def run_cmd(cmd, options, working_dir="."):
   cmd_str = " ".join(cmd)
   if options.verbose_mode:
-    print "Running command '%s' in dir %s." % (cmd_str, working_dir)
+    print (f"Running command '{cmd_str}' in dir {working_dir}")
 
   proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=working_dir)
   proc_stdout, proc_stderr = proc.communicate()
   ret_val = proc.wait()
+  proc_stdout_text = proc_stdout.decode('utf-8')
+  proc_stderr_text = proc_stderr.decode('utf-8')
 
   if options.debug_mode:
-    print "==== %s stdout start ====" % cmd_str
-    print proc_stdout
-    print "==== %s stdout end ====" % cmd_str
-    print "==== %s stderr ====" % cmd_str
-    print proc_stderr
-    print "==== %s stderr ====" % cmd_str
+    print (f"==== {cmd_str} stdout start ====")
+    print (f"{proc_stdout_text}")
+    print (f"==== {cmd_str} stdout end ====")
+    print (f"==== {cmd_str} stderr start ====")
+    print (f"{proc_stderr_text}")
+    print (f"==== {cmd_str} stderr end ====")
```

Comment:
`proc_{stdout,stdin}_text` are already strings, there is no need to use an extra f-string
```suggestion
    print (f"==== {cmd_str} stdout start ====")
    print (proc_stdout_text)
    print (f"==== {cmd_str} stdout end ====")
    print (f"==== {cmd_str} stderr start ====")
    print (proc_stderr_text)
    print (f"==== {cmd_str} stderr end ====")
```

Reply:
thanks, good catch, addressed in [0b3ca5e](https://github.com/kokkos/kokkos/pull/8720/commits/0b3ca5efa433c406e56f829f3e5285045190c859)